internal/telemetry/tracker.go

```diff
@@ -96,8 +119,369 @@ func (t *Tracker) trackClusterDeploymentHeartbeat(ctx context.Context) error {
+					if isGPUResource(resourceName) {
+						podAllocGPU = true
+						break
+					}
+				}
+			}
+		}
+
+		if podRequestGPU {
+			nodePodMetrics[pod.Spec.NodeName]["nodePodGpuRequestCount"]++
+		}
+		if podAllocGPU {
+			nodePodMetrics[pod.Spec.NodeName]["nodePodGpuAllocCount"]++
+		}
+	}
+
+	return nodePodMetrics, nil
+}
+
+// Helper function to build node-level metrics
+// Return a map indexed by node name, plus cluster-level aggregates
+func buildNodeLevelMetrics(ctx context.Context, k8sClient client.Client) ([]map[string]any, map[string]any, error) {
+	// Get all nodes in the cluster
+	nodes := &corev1.NodeList{}
+	if err := k8sClient.List(ctx, nodes); err != nil {
+		return nil, nil, err
+	}
+
+	clusterMetrics := make(map[string]any)
+	nodeMetrics := make([]map[string]any, 0)
+	controllerNodes := 0
+	workerNodes := 0
+	totalPods := 0
+	totalContainers := 0
+	totalPodsRequestGPU := 0
+	totalPodsAllocGPU := 0
+
+	// Count non-system pods per node
+	nodePodCounts, err := countNonSystemPodsOnNodes(ctx, k8sClient)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Iterate through nodes and build a node-level metrics map
+	for i, node := range nodes.Items {
+		if i == 0 {
+			clusterMetrics = map[string]any{
+				"kubernetesVersion": node.Status.NodeInfo.KubeletVersion,
+				"containerRuntime":  node.Status.NodeInfo.ContainerRuntimeVersion,
+			}
+		}
+
+		// Node CPU cores & Memory
+		cpuCores := node.Status.Capacity.Cpu().Value()
+		memoryBytes := node.Status.Capacity.Memory().Value()
+		memoryGB := memoryBytes / (1024 * 1024 * 1024) // Convert to GB
+
+		// Node Storage (ephemeral-storage)
+		diskBytes := int64(0)
+		if storage, exists := node.Status.Capacity["ephemeral-storage"]; exists {
+			diskBytes = storage.Value()
+		}
+		diskGB := diskBytes / (1024 * 1024 * 1024) // Convert to GB
+
+		// NVIDIA: Node Feature Discovery (NFD) and GPU Feature Discovery (GFD)
+		// See https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/gpu-sharing.html for more details on GPU sharing
+		gpuNvidiaCapacity := node.Status.Capacity[corev1.ResourceName("nvidia.com/gpu")]
+		gpuNvidiaCapacityShared := node.Status.Capacity[corev1.ResourceName("nvidia.com/gpu.shared")]
+		gpuNvidiaAlloc := node.Status.Allocatable[corev1.ResourceName("nvidia.com/gpu")]
+		gpuNvidiaAllocShared := node.Status.Allocatable[corev1.ResourceName("nvidia.com/gpu.shared")]
+
+		// Labels provide actual values for GPU counts, types, memory, and sharing configuration
+		gpuNvidiaCount := node.Labels["nvidia.com/gpu.count"]
+		gpuNvidiaProduct := node.Labels["nvidia.com/gpu.product"]
+		gpuNvidiaMemory := node.Labels["nvidia.com/gpu.memory"]
+		gpuNvidiaSharing := node.Labels["nvidia.com/gpu.sharing"]
+
+		// Node GPU cores and vendor
+		gpuCount := int64(0)
+		gpuType := "none"
+
+		// Scan resources for strings suggesting a GPU
+		for resourceName, quantity := range node.Status.Capacity {
+			if isGPUResource(resourceName) {
+				gpuCount += quantity.Value()
```

Comment:
i see the inconsistencies, we count gpu in bytes, memory and storage in GiB. is it expected?